doc/SAI-Proposal-PoE.md

```diff
@@ -0,0 +1,591 @@
+| PD             | Powered Device, can power the device through an PD-capable Ethernet port                          |
+| PoE Port       | Ethernet port that can deliver power and info to devices, located on the front panel of the board |
+
+# Key Ideas behind PoE
+
+1. PoE provides a hardware solution for providing a data connection and electricity to power connected devices
+2. The PoE software provides an interface to configure (program) and query the PoE controller and its internal peripherals such as ports etc
+3. Users can see PoE information and configure PoE features according to their needs
+
+# PoE Hardware Design Example
+
+![poe-physical-design](figures/PoE_Physical_Design.png)
+
+# SAI POE Components
+
+## New PoE SAI objects
+
+Three new SAI objects are introduced.
+
+### PoE Device ID
+
+This object is used to hold the global configuration and PoE hardware status. In many cases, one PoE device instance will have access to all the PoE hardware and components.
+
+### PoE PSE ID
+
+This object is used to access data from the Power Sourcing Equipment devices in the board. Such as PSE temperatures, versions, and statuses. There can be multiple PSEs in one board.
+
+### PoE Port ID
+
+This object is used to access data on a port level. PoE ports supply power using an ethernet cable. PoE port object gives access to read data and configure these ports.
+
+## New SAI Switch Type PoE
+
+A new abstraction type for PoE is created which allows multiple vendors to implement their own PoE software and hardware solutions.
+
+```
+/**
+ * @brief Attribute data for #SAI_SWITCH_ATTR_TYPE
+ */
+typedef enum _sai_switch_type_t
+{
+    /** Switch type is Switching Network processing unit */
+    SAI_SWITCH_TYPE_NPU,
+
+    /** Switch type is PHY */
+    SAI_SWITCH_TYPE_PHY,
+
+    /** Switch type is VOQ based NPU */
+    SAI_SWITCH_TYPE_VOQ,
+
+    /** Switch type is Fabric switch device */
+    SAI_SWITCH_TYPE_FABRIC,
+
+    /** Switch type is PoE (Power over Ethernet) */
+    SAI_SWITCH_TYPE_POE,
+
+} sai_switch_type_t;
+```
+
+## Example of how to initialize  PoE
+
+An example of how to initlizate PoE and access its components.
+
+### Create a PoE Device Object
+
+The PoE device object is used to access and configure board-wide system information and operations. Information about the kind of PoE hardware should be provided on the NOS level and used to initalize the driver.
+
+```
+    /**
+     * @brief Device Information for switch initialization.
+     *
+     * Hardware information format is based on POE SAI implementations by vendors.
+     * Format is vendor specific.
+     * Example: Like PCI location, I2C address, UART etc.
+     * In case of NULL, use the default PoE associated switch id.
+     *
+     * @type char
+     * @flags CREATE_ONLY
+     * @default ""
+     */
+    SAI_POE_DEVICE_ATTR_HARDWARE_INFO = SAI_POE_DEVICE_ATTR_START,
```

Comment:
Does this attribute SAI_POE_DEVICE_ATTR_HARDWARE_INFO refer to initializing the PoE controller ? If so, it is very misleading.

Reply:
The SAI_POE_DEVICE_ATTR_HARDWARE_INFO refers to initializing the PoE subsystem according to the PoE hardware.
For example, if the board has a MCU you can set the hw_info to "integrated_mcu". If the device uses direct i2c access then you can set the hw_info to "i2c" or an address. You can also send an empty string if this is not relevant for the sytem. 
This allows the driver to understand how to communicate with the PoE hardware.